### PR TITLE
add Minus operator

### PR DESCRIPTION
**Pull request by m-schen**
**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue #4094 

**What this PR does / why we need it:**

Not Available

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

File: pkg/sql/colexec/minus/minus.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+	arg := argument.(*Argument)
+	{
+		arg.ctr.bat = nil
+		arg.ctr.hashTable = hashmap.NewStrMap(true)
+		for i := range arg.ctr.scales {
+			arg.ctr.scales[i] = make([]int32, len(arg.Precisions[0]))
+		}
+	}
+	{
+		// aligning the decimal precision
```
Comment (fengttt) [inline pkg/sql/colexec/minus/minus.go:40]: Decimal does not need to be aligned any more.

**Review by fengttt — CHANGES_REQUESTED** [review]
Can please you comment in minus.go, describe what you do.   How do you do bookkeeping in the hashtable, etc.

This kind a change should have a bvt test case.

**Comment by m-schen** [comment]
> Can please you comment in minus.go, describe what you do. How do you do bookkeeping in the hashtable, etc.
> 
> This kind a change should have a bvt test case.

I have described how I did it in the comments of the call function.  
i will make a doc to describe how to use the hashtable next week.
this pr can not have bvt test because it is just an operator implement but without plan build. it is hard to do test end to end.

File: pkg/sql/colexec/minus/minus.go [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+		count := vector.Length(bat.Vecs[0])
+		itr := ctr.hashTable.NewIterator(arg.IBucket, arg.NBucket)
+		for i := 0; i < count; i += hashmap.UnitLimit {
+			oldHashGroup := ctr.hashTable.GroupCount()
+
+			n := count - i
+			if n > hashmap.UnitLimit {
+				n = hashmap.UnitLimit
+			}
+			vs, _ := itr.Insert(i, n, bat.Vecs, ctr.scales[index])
```
Comment (nnsgmsone) [inline pkg/sql/colexec/minus/minus.go:167]: The likelihood of doing so would be very low and I would advise against it.

**Comment by m-schen** [comment]
I've added comments to the minus.go file to describe how I do the minus work and removed the useless scalar related work. And bvt test can't be done at now because it is just an operator implement but without plan built.

I will push a hash table related doc today but not this pull request.

**Comment by m-schen** [comment]
I will make another pull request first, to reduce the cost of learning to use hash tables.
and change related codes about the use of hash table at this pr.

**Comment by m-schen** [comment]
> I want to make another pull request first, to reduce the cost of learning to use hash tables.
> and change related codes about the use of hash table at this pr.

**Pull request closed without merging**